lib/semantics/semantics.cc

```diff
@@ -83,7 +84,7 @@ using StatementSemanticsPass1 = ExprChecker;
 using StatementSemanticsPass2 = SemanticsVisitor<AllocateChecker,
     ArithmeticIfStmtChecker, AssignmentChecker, CoarrayChecker,
     DeallocateChecker, DoChecker, IfStmtChecker, IoChecker, NullifyChecker,
-    ReturnStmtChecker, StopChecker>;
+    OmpClauseChecker, ReturnStmtChecker, StopChecker>;
```

Comment:
Could this be an OmpChecker that does all semantics checking for all OpenMP directives?  A lot of that is directly associated with clauses, but there are also other restrictions such as:

 (4.5 page 62) _All loops associated with the loop construct must be perfectly nested_ 

 (4.5 page 90) _A program that branches into or out of a taskloop region is non-conforming._ 

(4.5 page 227) _Nesting of Regions_ restrictions.

So a checker might have Enter/Leave functions for directives that are integrated with Enter/Leave functions that process clauses for the current directive to validate not only clauses, but all other OpenMP restrictions.

Reply:
Actually, when I first thought about it, it was named as `OmpChecker` but to make it more specific, I re-named it later. Since you asked and that was a good point (most of the restrictions are directly associated with directives/clauses), I will put more thought into it, thanks!

Reply:
I decided to rename it to `OmpStructureChecker` since all the checks will be done in this `checker` are directly related to the validity of structures (directives/clauses).